sdk/communication/communication-chat/review/communication-chat.api.md

```diff
@@ -37,6 +37,16 @@ export interface AddParticipantsRequest {
     participants: ChatParticipant[];
 }
 
+// @public
+export interface ChatAttachment {
+    attachmentType: "teamsInlineImage";
```

Comment:
This isn't right. It shouldn't be typed to this string. It should just be a type of `string` or `AttachmentType`

Reply:
no that's incorrect. I have checked with JS architect and this is what we should be using

Reply:
There are 2 scenarios in which enums can be used:

1. Closed options - This is when the service only accepts a set of values. If the client passes a value that is not in this set the service fails. For this option, the preferred approach is string literal unions, like it is being done here.

2. Extensible options - This is when there is a set of commonly used values, however, the service can accept pretty much any value and it won't result in a failure. In this scenario, the "enum" is like a suggestion of possible values but not limited to them. In this scenario, the best we can do with Typescript is type the property as a string and expose a convenience enum for users that choose to use it.

What does the future look for this feature? I imagine we plan on supporting new types of attachments?

Reply:
thanks @joheredi ! we would have file attachment in near future so a new enum named 'file' would be added. Based on what I have talked with other folks, the current plan is to use string literal for single enum like this (since when it compiles to JS, it would not generate boilerplate code), then we would move to type enum when we work on file attachments. So it would be like this:

`
export declare type AttachmentType = "teamsInlineImage" | "file";

export interface ChatAttachment {
     attachmentType: AttachmentType;
     ...
}
`